DcCore/DcCore/Helper/DcUtils.swift

```diff
@@ -131,4 +131,15 @@ public struct DcUtils {
           return String.localized("connectivity_not_connected")
         }
     }
+
+    public static func showRecentlySeen(context: DcContext, chat: DcChat) -> Bool {
+        var recentlySeen = false
+        if !chat.isSelfTalk && !chat.isGroup && !chat.isMailinglist && !chat.isDeviceTalk {
+            let contactIds = chat.getContactIds(context)
+            if contactIds.count == 1 {
+                recentlySeen = context.getContact(id: contactIds[0]).wasSeenRecently
+            }
+        }
```

Comment:
on android, we discussed a bit how [get rid of the two additional database calls](https://github.com/deltachat/deltachat-android/pull/2371#discussion_r953833784) - and finally did by caching already loaded values.

however, i'd say unless we encounter some slowness or it is really easy to improve, we can go for it as is.